### ISSUE TITLE
Lightsaber textures don't load

### ISSUE THREAD

**Issue by Desmords** [elBukkit/MagicPlugin#808]
I have a problem with the plugin magic 8.1 server-1.12.2. When you turn on the lightsaber is not loading the texture. How can I solve this bug?

![image](https://user-images.githubusercontent.com/37225172/101099246-5aa4fb80-35d5-11eb-84b7-a81469657e44.png)
![image](https://user-images.githubusercontent.com/37225172/101099259-60024600-35d5-11eb-9910-304a39b79142.png)

**Comment by NathanWolf** [comment]
Whoops, sorry.

I will need to fix this in the next release, otherwise in the meantime you would have to fix each lightsaber color individually, I think 

`/mconfig configure wand lightsaber_blue legacy_icon shield_icon:3` 

would work for instance. If you want to go that route I can help out with the rest of the colors. 

Otherwise if you are willing to use a stable dev build I could have a fix ready pretty quickly.

**Comment by Desmords** [comment]
> Otherwise if you are willing to use a stable dev build I could have a fix ready pretty quickly.

Yes, make a stable build

**Comment by NathanWolf** [comment]
Please try this, if you can- I don't have easy access to a legacy MC server right now to test it, but I think it should fix the problem. Let me know if not.

https://jenkins.elmakers.com/job/Magic-Stable/19/artifact/Magic/target/Magic-8.1.1-SNAPSHOT.jar

**Comment by Desmords** [comment]
still doesn't work

**Comment by NathanWolf** [comment]
Ok, I will have to get back to you then.

**Comment by NathanWolf** [comment]
I think that last fix should've worked for newly created sabers, but for any of them that got accidentally migrated to the new icons it would not.

I've added some reverse-migration config in this one, I tested it on 1.12 with a new saber and an old one that I let get migrated and it seems OK now.

https://jenkins.elmakers.com/job/Magic-Stable/20/artifact/Magic/target/Magic-8.1.1-SNAPSHOT.jar

**Comment by Desmords** [comment]
Thank you!

**Comment by Desmords** [comment]
![image](https://user-images.githubusercontent.com/37225172/101194096-94761080-366e-11eb-966e-758aed665b4f.png)
Still doesn't work again

**Comment by NathanWolf** [comment]
Can you please show me the output of `/magic`?

Does it work if you get a new saber with `/mgive lightsaber_blue`?

**Comment by Desmords** [comment]
![image](https://user-images.githubusercontent.com/37225172/101194334-ec147c00-366e-11eb-950f-c5afc5d58a9e.png)
Only the blue lightsaber is activated. The rest are like shields.

**Comment by NathanWolf** [comment]
Is it just the blue one, or just new ones? If you do `/mgive lightsaber_green` does it work?

The first fix didn't work for old sabers, but I thought I fixed that in the second fix (which you do have, thank you for the version info)

**Comment by Desmords** [comment]
![image](https://user-images.githubusercontent.com/37225172/101197065-ac4f9380-3672-11eb-87fb-02d7438d5ba3.png)
Green lightsaber

**Comment by NathanWolf** [comment]
That seems really weird- I wonder if it's because of this, did you run this command when I asked?

`/mconfig configure wand lightsaber_blue legacy_icon shield_icon:3`

I guess that could be why only the blue one works if so ... but those changes should be in the new version for all sabers. Unless you've customized them all?  

I guess if that worked you could try the same for the rest of them

```
/mconfig configure wand lightsaber_green legacy_icon shield_icon:2
/mconfig configure wand lightsaber_red legacy_icon shield_icon:4
/mconfig configure wand lightsaber_orange legacy_icon shield_icon:5
/mconfig configure wand lightsaber_purple legacy_icon shield_icon:6
/magic load
```

Sorry for all the trouble, I'm not sure what could be wrong at this point, I'd like to fix it for the next release but if those commands fixes you up I don't want to take up any more of your time.

**Comment by Desmords** [comment]
When re-joining the lightsaber still becomes a shield

**Comment by NathanWolf** [comment]
I'm kind of at a loss. I've tested the fixes myself and it all seems to work.

If you want, try shutting down and deleting the entire `plugins/Magic` folder. If this is not a new server and you have progress you want to keep, then save `plugins/Magic/data`

Start back up and I'm hoping it is fixed. If not then I really don't know :(

**Comment by NathanWolf** [comment]
Wait, no sorry- hang on please

**Comment by NathanWolf** [comment]
Ah, nevermind- I thought I had found an issue when I reset my configs, but I had just forgotten to add the example back in.

So yes if you are just trying this out I'd suggest starting over if that's possible. 

Shut down, delete `plugins/Magic` (save `plugins/Magic/data` if you have progress).
Then `/mconfig example set stars` to to get the lightsabers back.

Make sure you're using the RP the plugin gives you, you don't have one set in your client nor server.properties

I've tested all this myself, I really don't know what else could be wrong at this point.
